kube/src/api/metadata.rs

```diff
@@ -11,17 +11,15 @@ use serde::{Deserialize, Serialize};
 /// And these optional properties:
 /// - .metadata.namespace
 /// - .metadata.resource_version
-///
-/// This avoids a bunch of the unnecessary unwrap mechanics for apps
 pub trait Meta: Metadata {
     /// Metadata that all persisted resources must have
     fn meta(&self) -> &ObjectMeta;
     /// The name of the resource
-    fn name(&self) -> String;
+    fn name(&self) -> &String;
     /// The namespace the resource is in
-    fn namespace(&self) -> Option<String>;
+    fn namespace(&self) -> &Option<String>;
     /// Tthe resource version
-    fn resource_ver(&self) -> Option<String>;
+    fn resource_ver(&self) -> &Option<String>;
```

Comment:
Should be `&str` and `Option<&str>`, respectively. That should cut down on the awkward `as_ref()`s, and `Option<&T>` is covered by the null pointer optimization.

Reply:
Pushed a change now that tries to do that signature. It shifts things a lot, but it doesn't feel any less awkward. If anything it feels more awkward internally.